tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -424,6 +424,204 @@ bool IsDefaultBranch(string candidate) => !StringUtil.IsNullOrWhiteSpace(default
+            if (gitLogDataArray.Length < 8)
+            {
+                Log.Warning<int>("GitCommandHelper.FetchCommitData: unexpected git show output – expected ≥ 8 tokens, got {Count}", gitLogDataArray.Length);
+                return null;
+            }
+
+            // Parse author and committer dates from Unix timestamp
+            if (!long.TryParse(gitLogDataArray[1], out var authorUnixDate))
+            {
+                Log.Warning("Error parsing author date from git log output");
+                return null;
+            }
+
+            if (!long.TryParse(gitLogDataArray[4], out var committerUnixDate))
+            {
+                Log.Warning("Error parsing committer date from git log output");
+                return null;
+            }
+
+            var commit = gitLogDataArray[0];
+            if (commit.StartsWith("'"))
+            {
+                commit = commit.Substring(1);
+            }
+
+            var commitMessage = string.Join("|,|", gitLogDataArray.Skip(7)).Trim();
+            if (commitMessage.EndsWith("'"))
+            {
+                commitMessage = commitMessage.Substring(0, commitMessage.Length - 1).Trim();
+            }
+
+            var commitData = new CommitData(
+                CommitSha: commit,
+                AuthorDate: DateTimeOffset.FromUnixTimeSeconds(authorUnixDate),
+                AuthorName: gitLogDataArray[2],
+                AuthorEmail: gitLogDataArray[3],
+                CommitterDate: DateTimeOffset.FromUnixTimeSeconds(committerUnixDate),
+                CommitterName: gitLogDataArray[5],
+                CommitterEmail: gitLogDataArray[6],
+                CommitMessage: commitMessage);
+
+            Log.Debug("GitCommandHelper.FetchCommitData: completed successfully for {Commit}", commitSha);
+            return commitData;
+        }
+        catch (Exception ex)
+        {
+            Log.Warning(ex, "GitCommandHelper.FetchCommitData: unexpected error while fetching data for {Commit}", commitSha);
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Determines whether the repository located at <paramref name="workingDirectory"/> is a shallow clone.
+    /// </summary>
+    /// <param name="workingDirectory">Path to the git working directory.</param>
+    /// <returns>True if the repository is a shallow clone; otherwise, false.</returns>
+    public static bool IsShallowCloneRepository(string workingDirectory)
+    {
+        // We need to check if the git clone is a shallow one before uploading anything.
+        // In the case is a shallow clone we need to reconfigure it to upload the git tree
+        // without blobs so no content will be downloaded.
+        var gitRevParseShallowOutput = RunGitCommand(
+            workingDirectory,
+            "rev-parse --is-shallow-repository",
+            MetricTags.CIVisibilityCommands.CheckShallow);
+        if (gitRevParseShallowOutput is null)
```

Comment:
Should you check for non-zero exit code here too?

Reply:
done in [c5102c4](https://github.com/DataDog/dd-trace-dotnet/pull/7285/commits/c5102c40f83d21ccbbcabc2d34f362b2045481b0)